controllers/slice/namespaces.go

```diff
@@ -718,9 +719,19 @@ func (r *SliceReconciler) labelAppNamespaces(ctx context.Context, cfgAppNsList [
 		// label does not exists on namespace
 		namespace := &corev1.Namespace{}
 		err := r.Get(ctx, types.NamespacedName{Name: cfgAppNs}, namespace)
+		// If the namespace itself doesn't exist on specified cluster then first create it
 		if err != nil {
-			log.Error(err, "Failed to find namespace", "namespace", cfgAppNs)
-			continue
+			log.Info("Failed to find namespace", "namespace", cfgAppNs)
+			namespace = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: cfgAppNs,
+				},
+			}
+			err2 := r.Create(ctx, namespace)
```

Comment:
Put this in a single if block to avoid "err2". 
``` 
if err := r.Create(ctx, namespace); err!=nil {
```

Reply:
Addressed